code/modules/surgery/plastic_surgery.dm

```diff
@@ -30,12 +30,42 @@
 		user.visible_message("[user] successfully restores [target]'s appearance!", "<span class='notice'>You successfully restore [target]'s appearance.</span>")
 	else
 		var/list/names = list()
+		var/list_size = 10
+		var/obj/item/card/id/ID
+
+		//IDs in hand
+		if(istype(user, /mob/living/carbon/human)) //Only 'humans' can hold ID cards
+			var/mob/living/carbon/human/H = user
+			ID = H.get_id_from_hands()
+			if(ID)
+				names += ID.registered_name
+				list_size -- //To stop list bloat
+
+		//IDs on body
+		var/list/ID_list = list()
+		for(var/obj/item/I in range(0, target)) //Get ID cards
+			if(istype(I, /obj/item/card/id))
+				ID_list += I
+			else if(istype(I, /obj/item/pda))
+				var/obj/item/pda/PDA = I
+				ID_list += PDA.id
+			else if(istype(I, /obj/item/storage/wallet))
+				var/obj/item/storage/wallet/W = I
+				ID_list += W.front_id
+		for(var/obj/item/card/id/Card in ID_list) //Add card names to 'names'
```

Comment:
Can be a typeless loop

Reply:
It needs to be set to an ID card, because of this
`ID = Card.registered_name`

Reply:
Typeless loops are the following:
```
var/list/L = list(new datum/stuff(), new datum/stuff())
for(var/thing in L)
 var/datum/stuff/S = thing
```
If you do a typed loop (the one you have now) it will filter the given list. Which is not required since you know that the list is only filled with a certain type.

Reply:
Ah, I see